src/gc/unix/cgroup.cpp

```diff
@@ -84,9 +66,113 @@ class CGroup
         free(mem_limit_filename);
         return result;
     }
+
+    bool GetCpuLimit(uint32_t *val)
+    {
+        char *cfs_period_filename = nullptr;
+        char *cfs_quota_filename = nullptr;
+        bool result = false;
+        long long quota;
+        long long period;
+        long long cpu_count;
+
+        if (m_cpu_cgroup_path == nullptr)
+            return false; 
+
+        cfs_period_filename = (char*)malloc(strlen(m_cpu_cgroup_path) + strlen(CFS_PERIOD_FILENAME) + 1);
+        if (cfs_period_filename == nullptr)
+            return false;
+
+        strcpy(cfs_period_filename, m_cpu_cgroup_path);
+        strcat(cfs_period_filename, CFS_PERIOD_FILENAME);
+        result = ReadLongLongValueFromFile(cfs_period_filename, &period);
+        free(cfs_period_filename);
+        if (!result || period <= 0)
+             return false;
+
+        cfs_quota_filename = (char*)malloc(strlen(m_cpu_cgroup_path) + strlen(CFS_QUOTA_FILENAME) + 1);
+        if (cfs_quota_filename == nullptr)
+            return false;
+
+        strcpy(cfs_quota_filename, m_cpu_cgroup_path);
+        strcat(cfs_quota_filename, CFS_QUOTA_FILENAME);
+        result = ReadLongLongValueFromFile(cfs_quota_filename, &quota);
+        free(cfs_quota_filename);
+        if (!result || quota <= 0)
+            return false;
+        
+        cpu_count = quota / period;
+        if (cpu_count >= 1 && cpu_count < UINT32_MAX){
+            *val = cpu_count;
+        } else {
+            *val = 1;
+        }
+
+        return true;
+    }
     
 private:
-    char* FindMemoryHierarchyMount()
+    static bool IsMemorySubsystem(const char *strTok){
+        return strncmp("memory", strTok, 6) == 0 && strlen(strTok) == 6;
```

Comment:
A nit - it seems that just `strcmp("memory", strTok)` would be equivalent to the combination of `strncmp` and `strlen`.